src/types/multi-select.ts

```diff
@@ -1,16 +1,220 @@
-export interface MultiselectItem {
+import type { LabelAttributes } from './label'
+import { type PopoverAttributes } from './popover'
+export interface MultiselectItem<T extends string = string> {
   label: string
-  value: string
+  value: T
   key?: string
   selected?: boolean
   disabled?: boolean
   custom?: boolean
   group?: string
 }
 
-export interface MultiselectFilterFunctionParams {
-  items: MultiselectItem[]
+export interface MultiselectFilterFunctionParams<T extends string = string> {
+  items: MultiselectItem<T>[]
   query: string
 }
 
 export type DropdownFooterTextPosition = 'sticky' | 'static'
+
+/**
+ * @internal
+ */
+export interface MultiselectItemProps<T extends string> {
+  item: MultiselectItem<T>
+}
+
+/**
+ * @internal
+ */
+export interface MultiselectItemEmits<T extends string = string> {
+  'selected': [item: MultiselectItem<T>]
```

Comment:
```suggestion
  selected: [item: MultiselectItem<T>]
```

Let's only add quotes to necessary keys like others.